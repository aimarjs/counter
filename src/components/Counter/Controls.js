import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../Button';

const Controls = ({
  onAdd, onReset, onRemove, value,
}) => (
  <Wrapper>
    <Button title="Add" clickHandler={onAdd} />
    <Button title="Reset" clickHandler={onReset} disable={value === 0} />
    <Button title="Remove" clickHandler={onRemove} disable={value === 0} />
  </Wrapper>
);

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: 300px;
`;

Controls.propTypes = {
  onAdd: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired,
};

export default Controls;
