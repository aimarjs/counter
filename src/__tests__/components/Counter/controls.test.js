import React from 'react';
import renderer from 'react-test-renderer';

import Controls from '../../../components/Counter/Controls';

describe('Controls Component', () => {
  it('capturing snapshot of Controls', () => {
    const onAdd = () => {};
    const onRemove = () => {};
    const onReset = () => {};
    const value = 0;

    const renderedValue = renderer.create(
      <Controls
        value={value}
        onAdd={onAdd}
        onRemove={onRemove}
        onReset={onReset}
      />,
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
