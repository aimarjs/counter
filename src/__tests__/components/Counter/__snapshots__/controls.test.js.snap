// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Component capturing snapshot of Controls 1`] = `
<div
  className="sc-bwzfXH fSAIos"
>
  <button
    className="sc-bdVaJa bDsUlA"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
  <button
    className="sc-bdVaJa bDsUlA"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
  <button
    className="sc-bdVaJa bDsUlA"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</div>
`;
